web/annotation_layer_builder.js

```diff
@@ -95,18 +98,46 @@ class AnnotationLayerBuilder {
    *   annotations is complete.
    */
   async render(viewport, intent = "display") {
-    const [annotations, hasJSActions = false, fieldObjects = null] =
-      await Promise.all([
-        this.pdfPage.getAnnotations({ intent }),
-        this._hasJSActionsPromise,
-        this._fieldObjectsPromise,
-      ]);
-
-    if (this._cancelled || annotations.length === 0) {
+    if (this.div) {
+      // For consistency, ensure that both the `AnnotationLayer.render` and
+      // `AnnotationLayer.update` code-paths are asynchronous.
+      await deferred;
```

Comment:
Although, thinking about this, I'm not sure if we really need/want to keep that behaviour now!?